test/unit/emergency_banner/deploy_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+require_relative '../../../lib/emergency_banner/deploy'
+
+describe "Emergency Banner::Deploy" do
+  context "#run" do
+    should "create an emergency_banner hash in Redis with a campaign_class and a heading" do
+      Redis.any_instance.expects(:hmset).with(:emergency_banner,
+                                              :campaign_class, "black",
+                                              :heading, "A title",
+                                              :short_description, "",
+                                              :link, "",
+                                              )
+
+      EmergencyBanner::Deploy.new.run("black", "A title")
+    end
+
+    should "create an emergency_banner with a campaign_class, heading and short_description" do
+      Redis.any_instance.expects(:hmset).with(:emergency_banner,
+                                              :campaign_class, "black",
+                                              :heading, "A title",
+                                              :short_description, "A short description of the event",
+                                              :link, "",
+                                              )
+
+      EmergencyBanner::Deploy.new.run("black", "A title", "A short description of the event")
+    end
+
+    should "create an emergency_banner with a campaign_class, heading and a link" do
+      Redis.any_instance.expects(:hmset).with(:emergency_banner,
+                                              :campaign_class, "black",
+                                              :heading, "A title",
+                                              :short_description, "",
+                                              :link, "https://www.gov.uk"
+                                              )
+
+      EmergencyBanner::Deploy.new.run("black", "A title", "", "https://www.gov.uk")
+    end
+
+    should "create an emergency_banner with a campaign_class, heading, short_description and a link" do
+      Redis.any_instance.expects(:hmset).with(:emergency_banner,
+                                              :campaign_class, "black",
+                                              :heading, "A title",
+                                              :short_description, "A short description of the event",
+                                              :link, "https://www.gov.uk"
+                                              )
+
+      EmergencyBanner::Deploy.new.run("black", "A title", "A short description of the event", "https://www.gov.uk")
+    end
+  end
+end
```

Comment:
Do we need a specific test for the case of keys already being set and then overwritten? Or is that going too far and we are trusting Redis to do the right thing?